iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,460 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> that handles creating, updating, getting and deleting configurations
+    /// </summary>
+    public class ConfigurationsClient
+    {
+        private const string ConfigurationRequestUriFormat = "/configurations/{0}?{1}";
+        private const string ConfigurationsRequestUriFormat = "/configurations/?top={0}&{1}";
+        private const string ApplyConfigurationOnDeviceUriFormat = "/devices/{0}/applyConfigurationContent?" + ClientApiVersionHelper.ApiVersionQueryString;
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        private IHttpClientHelper _httpClientHelper;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected ConfigurationsClient()
+        {
+        }
+
+        internal ConfigurationsClient(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
+        {
+            _httpClientHelper = new HttpClientHelper(
+                connectionProperties.HttpsEndpoint,
+                connectionProperties,
+                ExceptionHandlingHelper.GetDefaultErrorMapping(),
+                s_defaultOperationTimeout,
+                transportSettings.Proxy,
+                transportSettings.ConnectionLeaseTimeoutMilliseconds);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration)
```

Comment:
To avoid duplicating this API, you can simply add an optional cancellation token to this method's parameter list:

```suggestion
        public virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken = default)
```

Reply:
With this change, you can delete the method

```csharp
public virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken)
```

Reply:
Then you can make that same change for every other method in this class that takes a cancellation token